ios/src/main/java/org/ergoplatform/ios/wallet/AddReadOnlyWalletViewController.kt

```diff
@@ -34,15 +35,31 @@ class AddReadOnlyWalletViewController : ViewControllerWithKeyboardLayoutGuide()
 
         val descLabel = Body1Label()
         descLabel.text = texts.get(STRING_INTRO_ADD_READONLY)
-        tvAddress = createTextField()
-        tvAddress.returnKeyType = UIReturnKeyType.Done
-        tvAddress.delegate = object : UITextFieldDelegateAdapter() {
-            override fun shouldReturn(textField: UITextField?): Boolean {
-                userPressedDone(uiLogic)
-                return super.shouldReturn(textField)
+        tvAddress = createTextField().apply {
+            returnKeyType = UIReturnKeyType.Done
+            delegate = object : UITextFieldDelegateAdapter() {
+                override fun shouldReturn(textField: UITextField?): Boolean {
+                    userPressedDone(uiLogic)
+                    return super.shouldReturn(textField)
+                }
             }
+
+            val scanIcon = UIImageView(getIosSystemImage(IMAGE_QR_SCAN, UIImageSymbolScale.Small))
+            scanIcon.tintColor = UIColor.label()
+            scanIcon.contentMode = UIViewContentMode.Center
+            val scanView = UIView(CGRect(0.0, 0.0, 35.0, 30.0))
+            scanView.addSubview(scanIcon)
```

Comment:
Let's start some unification of UI elements and encapsulate such kind of repetitive code behind a bunch of constructor functions.

Reply:
Yes, I agree. I usually move the code to UIUtils or something similar on second use (which is soon). Let's keep this unresolved until then.
But a remark regarding constructor: Using the constructor for building UI elements on iOS turned out to be unreliable in general. For some elements it is working, for others not. This is caused by how RoboVM initializes the underlying native objects and connects them with their JVM counterpart. This might explain some code that looks weird but it is necessary.

Reply:
Oh, I meant to say factory methods which allow to abstract construction of UI elements.